### PR TITLE
Enforce verification of CNAME in first step of custom domain and sync custom domain instructions between dashboard and docs

### PR DESCRIPTION
**Pull request by joshenlim**
Sync dashboard and docs regarding adding custom domains

- Enforce verification of CNAME record at first step (as per [docs](https://supabase.com/docs/guides/platform/custom-domains#configure-a-cname)) 
image

- Inform about replacing CNAME record with TXT in TXT verification step (as per [docs](https://supabase.com/docs/guides/platform/custom-domains#configure-txt-verification))
image

### CODE REVIEW

File: studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx [hunk 1]
```diff
@@ -14,30 +14,48 @@ import {
...
-    if (!projectRef) {
-      throw new Error('Project ref is required')
+    if (!ref) throw new Error('Project ref is required')
+
+    const cnameVerified = await verifyCNAME(values.domain)
+    if (!cnameVerified) {
+      return ui.setNotification({
+        category: 'error',
+        duration: 4000,
+        message: `Your CNAME record for ${values.domain} has yet to be verified - please check back again in a bit.`,
```
Comment (darora) [inline studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx:52]: `has yet to be verified` seems to be imply that the issue is on our end, whereas in reality they might not have set the CNAME record as expected.

We should say we expect it to be X, but it was Y (or missing). Also, would not auto-dismiss the error notification.

**Comment by darora** [comment]
Can't tell if this is being done, but on the CLI (and in the docs), we also verify that they've switched back from the TXT record to CNAME before the final Activate step is executed https://supabase.com/docs/guides/platform/custom-domains#activate-your-domain

**Comment by joshenlim** [comment]
> Can't tell if this is being done, but on the CLI (and in the docs), we also verify that they've switched back from the TXT record to CNAME before the final Activate step is executed https://supabase.com/docs/guides/platform/custom-domains#activate-your-domain

this calls the custom-hostnames/activate endpoint which seems like it does the verification internally i think

File: studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx]: gotcha - updated the language here

File: studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx [hunk 2]
```diff
@@ -14,30 +14,47 @@ import {
...
+const CustomDomainsConfigureHostname = () => {
   const { ui } = useStore()
+  const { ref } = useParams()
+  const { mutateAsync: createCustomDomain } = useCustomDomainCreateMutation()
+  const { services } = useProjectSettings(ref as string | undefined)
+
+  // Get the API service
+  const API_SERVICE_ID = 1
+  const apiService = services ? services.find((x: any) => x.app.id == API_SERVICE_ID) : {}
+  const endpoint = apiService?.app_config?.endpoint
```
Comment (alaister) [inline studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx:30]: Are you able to use the new `useProjectSettingsQuery()` hook here? It'll make this all type safe :)

Here are a couple of usage examples that should be helpful here:

https://github.com/supabase/supabase/blob/09dce53d1a919226bb6d1a3415d2ad377115a3a9/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainConfig.tsx#L23-L25

https://github.com/supabase/supabase/blob/09dce53d1a919226bb6d1a3415d2ad377115a3a9/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainVerify.tsx#L137

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**